src/devices/secplus_v2.c

```diff
@@ -0,0 +1,459 @@
+
+        // break if we've recived both halfs
+        if (fixed_1.bits_per_row[0] > 1 && fixed_2.bits_per_row[0] > 1) {
+            break;
+        }
+
+    }
+
+    // Do was have what we need ??
+    if (fixed_1.bits_per_row[0] == 0 || fixed_2.bits_per_row[0] == 0) {
+        //  No?  Awww F'ck it then
+        if (decoder->verbose)
+            fprintf(stderr, "%s: DECODE_FAIL_SANITY\n", __func__);
+        return DECODE_FAIL_SANITY;
+    }
+
+    // Assemble rolling_1[] and rolling_2[] into rolling_digits[]
+    uint8_t rolling_digits[24] = {0};
+    uint8_t *r;
+
+    r    = rolling_digits;
+    *r++ = rolling_2[8];
+    *r++ = rolling_1[8];
+    for (int i = 4; i < 8; i++) {
+        *r++ = rolling_2[i];
+    }
+    for (int i = 4; i < 8; i++) {
+        *r++ = rolling_1[i];
+    }
+
+    for (int i = 0; i < 4; i++) {
+        *r++ = rolling_2[i];
+    }
+    for (int i = 0; i < 4; i++) {
+        *r++ = rolling_1[i];
+    }
+
+    // compute rolling_total from rolling_digits[]
+    uint32_t rolling_total = 0;
+    uint32_t rolling_temp  = 0;
+
+    for (int i = 0; i < 18; i++) {
+        rolling_temp = (rolling_temp * 3) + rolling_digits[i];
+        // fprintf(stderr, ">> %12d\t%d\n", rolling_temp, rolling_digits[i]);
+    }
+
+    // value is 28 bits thus need to shift over 4 bit
+    rolling_total = reverse32(rolling_temp);
+    rolling_total = rolling_total >> 4;
+
+
+    // delete this soon
+    if (decoder->verbose > 2) {
+        fprintf(stderr, "%s : rolling_temp = %u\n", __func__, rolling_temp);
+
+        fprintf(stderr, "\n\nrolling_temp : ");
+        for (int i = 0; i < 32; i++) {
+            fprintf(stderr, "%d", ((rolling_temp >> i) & 0x01));
+        }
+        fprintf(stderr, "\n\n");
+
+        fprintf(stderr, "rolling_total = %u\n", rolling_total);
+
+        fprintf(stderr, "\n\nrolling_total : ");
+        for (int i = 0; i < 32; i++) {
+            fprintf(stderr, "%d", ((rolling_total >> i) & 0x01));
+        }
+        fprintf(stderr, "\n\n");
+    }
+
+
+    // Assemble "fixed" data part
+    uint64_t fixed_total = 0;
+    uint8_t *bb;
+    bb = fixed_1.bb[0];
+    fixed_total ^= ((uint64_t)bb[0]) << 32;
+    fixed_total ^= ((uint64_t)bb[1]) << 24;
+    fixed_total ^= ((uint64_t)bb[2]) << 16;
+
+    bb = fixed_2.bb[0];
+    fixed_total ^= ((uint64_t)bb[0]) << 12;
+    fixed_total ^= ((uint64_t)bb[1]) << 4;
+    fixed_total ^= (bb[2] >> 4) & 0x0f;
+
+    // fprintf(stderr, "fixed_total = %lu\n", fixed_total);
+
+    // int button    = fixed_total >> 32;
+    // int remote_id = fixed_total & 0xffffffff;
+    char fixed_str[16];
+    char rolling_str[16];
+
+    // rolling_total is a 28 bit unsigned number
+    // fixed_totals is 40 bit in a uint64_t
+    snprintf(fixed_str, sizeof(fixed_str), "%lu", fixed_total);
+    snprintf(rolling_str, sizeof(rolling_str), "%u", rolling_total);
+
+    /* clang-format off */
+    data_t *data;
+    data = data_make(
+            "id",       "ID",    DATA_STRING, "Secplus_v2",
```

Comment:
Change this to `"Security+ 2.0"`?

Reply:
should be "model" and "Secplus-v2" would be prefered.

Reply:
done

Reply:
done